03-Lab/SpatialImageProcessing.ipynb

```diff
@@ -0,0 +1,534 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Lab 03: Spatial image processing\n",
+    "\n",
+    "The purpose of this lab is to practice preprocessing needed on T1 images images and spatial normalization. Although spatial normalization is needed for the functional images as well, we will first get used to the concept with structural images."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Preparation\n",
+    "* import python tools for plotting 3D images\n",
+    "* copy basic T1 images from our osfshare for easy access to the files for this lab"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from nilearn import plotting\n",
+    "from nilearn import image"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "# Copy files from lab 01 for easier work with them here\n",
+    "# Name them generic names for ease of seeing how the command line programs work\n",
+    "cp ../01-Lab/osfshare/sub-97-T1w_brain.nii.gz my_T1_brain.nii.gz\n",
+    "cp ../01-Lab/osfshare/sub-97-T1w_defaced.nii.gz my_T1.nii.gz"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### How did we get that \"brain\" image with the skull stripped  again?\n",
+    "* FSL provides a tool called _bet_ which stands for _brain extraction tool_. See the [bet user manual here](https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/BET/UserGuide). \n",
+    "* It requires only the brain image with skull as input and has relatively few options. Performance is generally OK, but it is common to need to tweak the -f and -g options. \n",
+    "* If you want to get serious about getting a great brain mask (e.g., your brains have many individual differences in shape due to atrophy), you should look to [ANTs](http://stnava.github.io/ANTs/). Here is an [example](https://dpaniukov.github.io/2016/06/06/brain-extraction-with-ants.html) of using their brain extraction tool\n",
+    "* *Good brain extraction is essential to good registration*. If you see poor registration of your images later on then one place to start trouble-shooting is to look at the quality of your brain extraction by comparing the brain mask (what the program thinks is brain) to the image with the skull (did the program include non-brain tissues in the mask OR chop off part of the brain?)\n",
+    "    - Never assume BET worked great, always check!!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# refresher, are my images here?\n",
+    "!ls"
```

Comment:
I'm going to switch these blocks to `%%bash`. my reasoning is that while `!<bash-command>` is a great shortcut, it may confound the python and bash environments, which we are using interchangeably in these notebooks. For that reason, i think being explicit about the  bash code blocks is best for the students to compartmentalize what language the command is written in.